File: 一星題/10035-Primary Arithmetic.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int a, b;
    while (cin >> a >> b && (a || b))
    {
        int count = 0, carry = 0;
        while (a || b)
        {
            if ((a % 10 + b % 10 + carry) >= 10)
            {
                count++;
                carry = 1;
            }
            else
                carry = 0;
            a /= 10;
            b /= 10;
        }
        if (count > 1)
            cout << count << " carry operations." << endl;
        else if (count == 1)
            cout << count << " carry operation." << endl;
        else
            cout << "No carry operation." << endl;
    }
}

File: 一星題(Java)/10035-Primary Arithmetic.java
import java.util.Scanner;

public class Main{
    public static void main(String[] args){
        Scanner keyin = new Scanner(System.in);
        
        
        while(keyin.hasNextInt()){
            int num1 = keyin.nextInt();
            int num2 = keyin.nextInt();

            if(num1 == 0 && num2 == 0) break;
            int count = 0, carry = 0;
            while(num1 > 0 || num2 > 0){
                if(num1 % 10 + num2 % 10 + carry >= 10){
                    carry = 1;
                    count++;
                }else{
                    carry = 0;
                }
                num1 /= 10;
                num2 /= 10;
            }

            if(count == 0){
                System.out.println("No carry operation.");
            }else if(count == 1){
                System.out.println("1 carry operation.");
            }else{
                System.out.println(count + " carry operations.");
            }
        }
        
    }
}
